sklearn/decomposition/nmf.py

```diff
@@ -305,8 +306,12 @@ def _initialize_nmf(X, n_components, init=None, eps=1e-6,
     check_non_negative(X, "NMF initialization")
     n_samples, n_features = X.shape
 
+    if init == 'nndsvd' and n_components > min(n_samples, n_features):
```

Comment:
The NNDSVD is performed also with `init = 'nndsvda'` and `init = 'nndsvdar'`, which also need the same constraint.
You can use `if init != 'random'` to handle all three cases.
Please also update your unit test accordingly.

Reply:
Good point. Thanks!